.travis.yml

```diff
@@ -21,33 +21,26 @@ cache:
 
 before_install:
   - export CXX=g++-4.8
-  - export GOPATH=$HOME/gopath
-  - export PATH=$GOPATH/bin:$PATH
-  - export CHROME_BIN=/usr/bin/google-chrome-beta
-  - export PATH=$PWD/node_modules/.bin:$PATH
-  - export PROTOC_VERSION=3.0.0-beta-2
   - env
   - npm install -g npm@3.5.3
   - node --version
   - npm --version
-  - go version
-  - ls $CHROME_BIN
   - $CXX --version
-  - ./scripts/setup_go
 
 before_script:
+  - source ./scripts/setup_env.sh
+  - ./scripts/setup_go
+  - ls $CHROME_BIN
+  - go version
   - tsc --version
   - tslint --version
   - tsfmt --version
-  - jspm config registries.github.timeouts.lookup 60
-  - jspm config registries.github.timeouts.build 120
-  - jspm config registries.github.remote https://github.jspm.io
-  - jspm config registries.github.auth $JSPM_GITHUB_AUTH_TOKEN
-  - jspm config registries.github.maxRepoSize 0
-  - jspm config registries.github.handler jspm-github
-  - jspm install
-  - tsd install
+  - ./scripts/setup_typescript_test_packages.sh
   - ./scripts/protobuf_install
-
-script: ./scripts/copy_typescript_dtss && ./scripts/ensure_clean && ./scripts/ensure_jsonfmt && ./scripts/tsfmt && ./scripts/tslint && ./scripts/build.sh && ./scripts/test_run.sh
-
+  - ./scripts/copy_typescript_dtss
+  - ./scripts/ensure_jsonfmt
+  - ./scripts/tsfmt
+  - ./scripts/tslint
+  - ./scripts/build.sh
+  - ./scripts/test_run.sh
```

Comment:
Write a script `test_runner.sh` that calls everything we want as part of the test run.

```bash
source "${BASH_SOURCE%/*}/common.sh"

set -x

./scripts/build.sh
# ...
```


```